tests/superphot_plus/diff_test.py

```diff
@@ -1,109 +1,261 @@
+    ----------
+    file_name : str
+        The file to compare (a file of this name should exist in each dir).
+    goldens_dir : pathlib Path
+        The directory containing our goldens.
+    temp_results_dir : pathlib Path
+        The directory containing the temporary results files we've just generated.
+
+    Returns
+    -------
+    boolean
+        True if files are suficiently similar; False otherwise."""
+
+    # Set up
+    deltas = {"dynesty": 0.0001, "svi": 0.01, "NUTS": 0.0001}
+    no_differences_found = True
+
+    # Compare file sizes
+    data_goldens = np.load(Path(goldens_dir, file_name))["arr_0"]
+    data_new_results = np.load(Path(temp_results_dir, file_name))["arr_0"]
+    if len(data_goldens) != len(data_new_results):
+        print(f"Mismatched number of rows in {file_name}.")
+        return False
+    if data_goldens.size != data_new_results.size:
+        print(f"Mismatched number of vals in {file_name}.")
+        return False
+
+    # Compare sample means
+    lightcurve_name = file_name.split("_")[0]
+    sampling_method = file_name.split("_")[-1].split(".")[0]
+
+    goldens_samples = PosteriorSamples.from_file(
+        input_dir=goldens_dir, name=lightcurve_name, sampling_method=sampling_method
+    )
+    temp_results_samples = PosteriorSamples.from_file(
+        input_dir=temp_results_dir,
+        name=lightcurve_name,
+        sampling_method=sampling_method,
+    )
+
+    for index, golden_val in enumerate(goldens_samples.sample_mean()):
+        temp_val = temp_results_samples.sample_mean()[index]
+        if not math.isclose(golden_val, temp_val, rel_tol=deltas[sampling_method]):
+            no_differences_found = False
+            print(
+                f"\nSignificantly distant values in sample means of {file_name} at index {index} "
+                f"(and possibly later)."
+                f"\nDelta (relative tolerance) of {sampling_method} = {deltas[sampling_method]}."
+                f"\nGoldens sample mean: {goldens_samples.sample_mean()}"
+                f"\nNew results sample mean: {temp_results_samples.sample_mean()}"
+                f"\n"
+            )
+            break
+
+    print(f"Checked {file_name} (delta={deltas[sampling_method]}).")
+
+    return no_differences_found
+
+
+def delete_temp_files(temp_dir):
+    """Deletes all files in the given directory, then deletes the directory itself.
+
+    Parameters
+    ----------
+    temp_dir : pathlib Path
+        The temporary directory where we've just generated our new results.
+    """
+    if "temp" not in temp_dir.as_posix():
+        raise ValueError("Attempted to delete directory that is not designated with 'temp'.")
+    for file in os.listdir(temp_dir):
+        os.remove(Path(temp_dir, file))
+    os.rmdir(temp_dir)
 
 
 def test_diffs(tmp_path):
-    """Make golden files if they do not yet exist; otherwise, make new
-    results and compare with goldens.
+    """Check new results against pre-generated goldens file.
```

Comment:
This isn't called anywhere? Should this be removed?

Reply:
This is the one that will be run by pytest (by virtue of it being in the tests/ directory and the function beginning with "test")

Reply:
Makes sense. Thanks